dev/TabView/TabView.cpp

```diff
@@ -223,9 +232,55 @@ void TabView::OnListViewGettingFocus(const winrt::IInspectable& sender, const wi
     }
 }
 
+void TabView::SetTabViewItemAdjacentState(const int index, const winrt::hstring& adjacentState)
+{
+    if (const auto tab = TabFromIndex(index).try_as<winrt::TabViewItem>())
+    {
+        bool visualStateChanged = winrt::VisualStateManager::GoToState(tab, adjacentState, false);
+        // If element was not yet loaded and added to the tree,
+        // GoToState will return false and visual state will not be changed.
+        // In order to set the visual state in this case, we attach a Loaded handler
+        // that will set the visual state and then will be removed.
+        if (!visualStateChanged)
+        {
+            auto tabViewItemRevokers = winrt::make_self<TabViewItemRevokers>();
```

Comment:
I think a more robust approach is just handling on load on the TVI and ask the TabView what to instead of injecting a loaded event that get's revoked.

Reply:
Also how would this deal with virtualization? What if the TVI get's recycled?

Reply:
OOP approach as in Object Oriented Programming? Don't think that that get's violated. Either way these components need to talk with each other.

Also does the TabView listen to recycling of elements? I wasn't able to find anything in this PR that would account for that, so if an element get's recycled TabView doesn't seem to be aware of that.

Reply:
I don't think TabView does. Please, correct me if I'm wrong, but underlying ListView should take a care of that, shouldn't it?:)

Reply:
Yes ListView handles recycling, but currently you specify the bottom border on an element based on the position. However if it get's recycled onto a different position, it doesn't have to be adjacent to the selected element anymore. So you could end up where the visuals would be out of sync since the actual position in relation to the selected element.

Reply:
I see. However, the bottom border (it's corner radius) is being dictated by certain Visual State. When the item will be recycled, will the Visual State that was set before not "nullify" (become unset)? :)

Reply:
So if they get nullified, you won't update again then right? Since TabView doesn't listen to element creation, how would it be aware of an item being shown next the selected index?

Reply:
The bottom strokes will be update on every selected index change :)

Reply:
But if you scroll the selected index out of the realization rect, you have lost the bottom stroke value since TabView now thinks "Yeah we set the bottom stroke, we are done". If you then scroll to the selected index, TabView doesn't update the stroke anymore correctly. So you could end up with wrongly set stroke values.

Reply:
I see what you mean, I'll look into it :)